rbac/internal/views.py

```diff
@@ -403,6 +403,55 @@ def invalid_default_admin_groups(request):
     return HttpResponse('Invalid method, only "DELETE" and "GET" are allowed.', status=405)
 
 
+def role_removal(request, role_uuid):
+    """View method for internal role removal requests.
+
+    DELETE /_private/api/utils/roles/<role_uuid>/
+    """
+    logger.info(f"Role removal: {request.method} {request.user.username}")
+    if request.method == "DELETE":
+        if not destructive_ok():
+            return HttpResponse("Destructive operations disallowed.", status=400)
+
+        role_obj = get_object_or_404(Role, uuid=role_uuid)
+        with transaction.atomic():
+            try:
+                logger.warning(f"Deleting role {role_uuid}. Requested by {request.user.username}")
+                role_obj.delete()
+                return HttpResponse(status=204)
+            except Exception:
+                return HttpResponse("Role cannot be deleted.", status=400)
+    return HttpResponse('Invalid method, only "DELETE" is allowed.', status=405)
+
+
+def permission_removal(request):
+    """View method for internal permission removal requests.
+
+    DELETE /_private/api/utils/permission/
+    """
+    logger.info(f"Permission removal: {request.method} {request.user.username}")
+    if request.method == "DELETE":
+        if not destructive_ok():
+            return HttpResponse("Destructive operations disallowed.", status=400)
+
+        permission = request.GET.get("permission")
+        if not permission:
+            return HttpResponse(
+                'Invalid request, must supply the "permission" query parameter.',
+                status=400,
+            )
+
+        permission_obj = get_object_or_404(Permission, permission=permission)
```

Comment:
`permission` is  permission name ? is it unique ? can specify tenant id for certitude ?